From 86695e4c931397ebb19ba6f47708cd04adeb4e4b Mon Sep 17 00:00:00 2001
From: Misael Lopez Cruz <misael.lopez@ti.com>
Date: Mon, 6 Jun 2011 19:54:42 -0500
Subject: [PATCH 53/54] ASoC: DSP: Set correct widget type for capture FE/BE

For capture stream direction, the FE type is AIF_OUT and
BE type is AIF_IN. This will allow to search the correct
type of widgets for audio capture.

Signed-off-by: Misael Lopez Cruz <misael.lopez@ti.com>
---
 sound/soc/soc-dsp.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/sound/soc/soc-dsp.c b/sound/soc/soc-dsp.c
index 0dd90d11..1770ee8 100644
--- a/sound/soc/soc-dsp.c
+++ b/sound/soc/soc-dsp.c
@@ -158,8 +158,8 @@ static int dsp_add_new_paths(struct snd_soc_pcm_runtime *fe,
 		fe_type = snd_soc_dapm_aif_in;
 		be_type = snd_soc_dapm_aif_out;
 	} else {
-		fe_type = snd_soc_dapm_aif_in;
-		be_type = snd_soc_dapm_aif_out;
+		fe_type = snd_soc_dapm_aif_out;
+		be_type = snd_soc_dapm_aif_in;
 	}
 
 	/* get number of valid playback paths and their widgets */
-- 
1.7.4.1

